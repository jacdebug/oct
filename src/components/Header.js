import React from 'react';
import styled from 'styled-components';
import logo from '../assets/oc-logo.png';
import Box from './Box';
import Links from './Links';

export const Header = ({ layout }) =>
  <HeaderTag>
    <Logo href="/" title="Oddschecker" />
    <Box>
      <Box>
        <Links links={layout.sportLinks} />
      </Box>
      <MenuBreak />
      <Box>
        <Links bright links={layout.headerLinks} />
      </Box>
    </Box>
  </HeaderTag>;

export default Header;

const HeaderTag = styled.header`
  border-top: 2px solid ${props => props.theme.tcolor3};
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  min-height: 70px;
`;

const Logo = styled.a`
  background-image: url(${logo});
  background-size: 220px;
  display: inline-block;
  margin: 0 0 0 10px;
  width: 220px;
  height: 45px;
`;

const MenuBreak = styled.div`
  border-right: 1px solid ${props => props.theme.tcolor3};
  height: 25px;
  margin: 0 10px;
  opacity: 0.5;
`;
