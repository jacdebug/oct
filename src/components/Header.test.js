import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './Header';

it('renders without crashing', () => {
  shallow(<Header layout={[]} />);
});

it('renders a snapshot', () => {
  const tree = shallow(<Header layout={[]} />);
  expect(toJson(tree)).toMatchSnapshot();
});
