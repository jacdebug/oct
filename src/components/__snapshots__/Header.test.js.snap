// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<styled.header>
  <styled.a
    href="/"
    title="Oddschecker"
  />
  <styled.div>
    <styled.div>
      <Links />
    </styled.div>
    <styled.div />
    <styled.div>
      <Links
        bright={true}
      />
    </styled.div>
  </styled.div>
</styled.header>
`;
